ospfd/ospf_zebra.c

```diff
@@ -1351,11 +1351,27 @@ static int ospf_distribute_list_update_timer(struct thread *thread)
 						default_refresh = 1;
 					else if (
 						(lsa = ospf_external_info_find_lsa(
-							 ospf, &ei->p)))
+							 ospf, &ei->p))) {
+						int force =
+							LSA_REFRESH_IF_CHANGED;
+						/* If this is a MaxAge LSA, we
+						 * need to force refresh it
+						 * because distribute settings
+						 * might have changed and now,
+						 * this LSA needs to be
+						 * originated, not be removed.
+						 * If we don't force refresh it,
+						 * it will remain a MaxAge LSA
+						 * because it will look like it
+						 * hasn't changed. Neighbors
+						 * will not receive updates for
+						 * this LSA.
+						 */
+						if (IS_LSA_MAXAGE(lsa))
+							force = LSA_REFRESH_FORCE;
 						ospf_external_lsa_refresh(
-							ospf, lsa, ei,
-							LSA_REFRESH_IF_CHANGED);
-					else
+							ospf, lsa, ei, force);
+					} else
```

Comment:
https://github.com/FRRouting/frr/pull/6981 same changes are there in both the commits.

Reply:
Although the result should be the same, the code is different. #6987 has duplication calls of `ospf_external_lsa_refresh()` where only the flag differs. Also, the comment for the case being addressed is missing there.

Reply:
@KaushikNiral can you please reslove the conflict for the fix being done here? Your PR also duplicating the same fix.

Reply:
Yes, sure.

Reply:
Hi @Spantik, 

Resolved the conflict and raised a new PR #7074 
Please take a look.